stetho/src/main/java/com/facebook/stetho/common/android/HandlerUtil.java

```diff
@@ -62,4 +130,56 @@ public void join() {
       }
     }
   }
+
+  private static final class WaitableCallable<V> extends AbstractWaitableRunnable {
```

Comment:
You can get away with writing less code by treating a Runnable as a Callable.  You're already allocating one extra object for the waitable version, might as well add another to make your code about half as large :)